### ISSUE TITLE
Assignment of lambda to delegate fails for value type with generic NRT

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#47167]
**Version Used**: 3.8.0-2.20408.4

**Steps to Reproduce**:

```
delegate void D<T>(out T? value);
D<int> d = (out int v) => v = 0;
```

**Expected Behavior**:

No warnings.

**Actual Behavior**:

> Warning CS8622: Nullability of reference types in type of parameter 'v' of 'lambda expression' doesn't match the target delegate D\<int>'

**Comment by RikkiGibson** [comment]
I didn't understand this bug. Because T is unconstrained, I expected the constructed signature to look like this:

```cs
delegate void D<int>(out int value);
```

in which case, `(out int v) => ...` is a compatible lambda signature.

**Comment by jcouv** [comment]
Fixed in 17.1
